### PR TITLE
Refactor test framework by using t.Cleanup()

### PR DESCRIPTION
**Pull request by dgrisonnet**
cc @rhobs/team-monitoring

### CODE REVIEW

File: cmd/kube-events-exporter/main.go [hunk 1]
```diff
@@ -77,12 +77,12 @@ func main() {
 	eventListenAddr := net.JoinHostPort(opts.Host, strconv.Itoa(opts.Port))
 	eventListener, err := net.Listen("tcp", eventListenAddr)
 	if err != nil {
-		klog.Fatalf("Could not start listening on: %v: %v", eventListenAddr, err)
+		klog.Fatalf("start listening on: %v: %v", eventListenAddr, err)
```
Comment (lilic) [inline cmd/kube-events-exporter/main.go:80]: maybe `failed starting listening on events address: `

File: test/framework/kube-events-exporter.go [hunk 2]
```diff
@@ -64,78 +68,58 @@ type KubeEventsExporter struct {
...
-	}
-
-	// Recreate kube-events-exporter pod.
-	err = f.UpdateDeploymentReplicas(f.Exporter.Deployment, 1)
-	if err != nil {
-		return errors.Wrapf(err, "update deployment %s replicas to 1", f.Exporter.Deployment.Name)
-	}
+// GetEventMetricFamilies gets metrics from the event server metrics endpoint
+// and converts them to Prometheus MetricFamily.
+func (kee *KubeEventsExporter) GetEventMetricFamilies() (map[string]*dto.MetricFamily, error) {
```
Comment (lilic) [inline test/framework/kube-events-exporter.go:112]: Nit pick: just `e` here maybe? kee reads strange :)

File: test/framework/deployment.go [hunk 3]
```diff
@@ -83,10 +84,13 @@ func (f *Framework) DeleteDeployment(ns, name string) error {
 
 // WaitUntilDeploymentReady waits until given deployment is ready.
 func (f *Framework) WaitUntilDeploymentReady(ns, name string) error {
-	err := wait.Poll(5*time.Second, f.DefaultTimeout, func() (bool, error) {
+	err := wait.Poll(time.Second, f.DefaultTimeout, func() (bool, error) {
```
Comment (lilic) [inline test/framework/deployment.go:87]: is retrying every second what you think works best, or what is the reason behind this change?

File: cmd/kube-events-exporter/main.go [hunk 4]
```diff
@@ -42,7 +42,7 @@ func main() {
 
 	err := opts.Parse()
 	if err != nil {
-		klog.Fatalf("Error: %s", err)
+		klog.Fatalf("parse options: %v", err)
```
[the remark this answers is not in the document]
Comment (dgrisonnet) [reply cmd/kube-events-exporter/main.go:45]: I omitted the error indication in the message because I found it redundant with the log level already indicating that a fatal error occurred. What do you think should be the best approach?

File: test/e2e/exporter_test.go [hunk 5]
```diff
@@ -22,38 +22,32 @@ import (
...
-			if err != nil {
-				t.Logf("Could not close response body: %v\n", err)
-			}
-		}
-
-		families, err := framework.GetExporterMetricFamilies()
+		err = resp.Body.Close()
 		if err != nil {
-			t.Fatalf("Could not get exporter metrics: %v\n", err)
+			t.Logf("close response body: %v", err)
```
[the remark this answers is not in the document]
Comment (dgrisonnet) [reply test/e2e/exporter_test.go:35]: Since I discard the content of the request body here, I think that failing to close this stream should not impact this test result.

File: test/framework/deployment.go (hunk 3 above)
Comment (dgrisonnet) [reply test/framework/deployment.go:87]: Yes. With the new changes, I keep recreating the exporter, thus I felt that I needed to reduce the polling interval in order to optimize the setup time of each test.

**Comment by dgrisonnet** [comment]
@lilic I addressed your comments, please have a new look at it :slightly_smiling_face:
